what-is-chalk.mdx

```diff
@@ -18,215 +18,467 @@ import { PyDiffEditor } from '@/components/Editor'
 
-Chalk has built-in support for feature engineering workflows --
-there's no need to manage Airflow or orchestrate complicated streaming flows.
-You can execute resolver pipelines with declarative caching,
-ingest batch data on a schedule, and easily make slow sources
-available online for low-latency serving.
+```py
+@online
+def get_username(
+    # the input type annotation specifies the required feature
+    email: User.email,
+    # the resulting feature is saved as username unto User class
+) -> User.username:
+    username = email.split("@")[0]
+    if "gmail.com" in email:
+        username = username.split("+")[0].replace(".", "")
+
+    return username.lower()
+```
+
+Although written in Python, this function compiles into a static Velox expression through Chalk's Symbolic Python Interpreter and executes natively in C++.
+This provides Python-first development without sacrificing sub-5ms inference latency.
+
+```py
+lower(
+    if_then_else(
+        !=(
+            strpos(str(email), str(gmail.com)),
+            int(0)
+        ),
+        replace(
+            element_at(
+                split(
+                    element_at(split(str(email), str(@)), int(1)),
+                    str(+)
+                ),
+                int(1)
+            ),
+            str(.),
+            str()
+        ),
+        element_at(split(str(email), str(@)), int(1))
+    )
+)
+```
+
+We wrote an [engineering blog post on our Symbolic Python interpreter](https://chalk.ai/blog/symbolic-python-interpreter) if you want to learn more!
 
-### Caching
+### How can I cache features in Chalk?
 
-Many data sources (like vendor APIs) are too slow for online use cases
-or charge a high dollar cost-per-call. Chalk lets you optimize latency
-and cost by defining declarative caching policies that are well-integrated
-throughout the system. You no longer have to manage data sources such as
-Redis, Memcached, or DynamoDB, or spend time tuning cache-warming pipelines.
+Some data sources, like LLMs or 3rd party APIs, are expensive or slow to call at inference time.
+Chalk supports caching features to optimize for latency and cost:
 
-Add a caching policy with one line of code in your feature definition:
+- define cache windows inline with your features
+- override staleness at query-time when fresh data is critical
+- pre-warm caches and backfill pipelines for performance
+
+Add a caching policy with one line of code in our feature definition:
 
 <PyDiffEditor html={addCachingCode} />
 
-Optionally warm feature caches by executing resolvers on a schedule:
-
-```sql
--- resolves: User
--- source: snowflake
--- cron: 1h
-select
-    user_id as id,
-    credit_report,
-    ts
-from credit_scores
+
+### How do I deploy and query features with Chalk?
+
+After defining pipelines, deploy features to a branch in < 1s with:
+Chalk automatically detects and uses your current git branch name without requiring you to explicitly specify it:
+
+```shell
+chalk apply --branch
```

Comment:
can we show what gets printed after?

Reply:
I tihnk this is still missing? I would expect to see something like "Branch deploy to ... succeeded!" and a checkmark or something

Reply:
It's there I just had text in between the command and the output of the command. I reduced the amount of text to make it more clear